public/js/modules.js

```diff
@@ -2468,6 +2468,109 @@ RGBLed.prototype.customDraw = function() {
     ctx.fill();
 }
 
+function SquareRGBLed(x, y, scope = globalScope, dir = "UP", pinLength = 1) {
+    CircuitElement.call(this, x, y, scope, dir, 8);
+    this.rectangleObject = false;
+    this.setDimensions(15, 15);
+    this.pinLength = pinLength === undefined ? 1 : pinLength;
+    var nodeX = -10 - 10 * pinLength;
+    this.inp1 = new Node(nodeX, -10, 0, this, 8, "R");
+    this.inp2 = new Node(nodeX, 0, 0, this, 8, "G");
+    this.inp3 = new Node(nodeX, 10, 0, this, 8, "B");
+    this.inp = [this.inp1, this.inp2, this.inp3];
+    this.labelDirection = "UP";
+    this.fixedBitWidth = true;
+    
+    this.changePinLength = function (pinLength) {
+        if (pinLength == undefined) return;
+        pinLength = parseInt(pinLength, 10);
+        if (pinLength < 0 || pinLength > 1000) return;
+
+        // Calculate the new position of the LED, so the nodes will stay in the same place.
+        var diff = 10 * (pinLength - this.pinLength);
+        var diffX = this.direction == "LEFT" ? -diff : this.direction == "RIGHT" ? diff : 0;
+        var diffY = this.direction == "UP" ? -diff : this.direction == "DOWN" ? diff : 0;
+
+        // Build a new LED with the new values; preserve label properties too.
+        var obj = new window[this.objectType](this.x + diffX, this.y + diffY, this.scope, this.direction, pinLength);
+        obj.label = this.label;
+        obj.labelDirection = this.labelDirection;
+
+        this.cleanDelete();
```

Comment:
Any reason why delete the object and create a new one instead of just moving the position of the current object?

Reply:
The reason I did it this way was to avoid having to manipulate the existing nodes, to keep them in the same position. Now I realize that deleting and recreating causes connecting wires to get deleted as well, so maybe it would be interesting to try the other way around and see if we can keep the wires connected when changing pin length.

Reply:
Yes, I agree. 
It is fine for now, it won't affect the working of the LEDs in any way. Please feel free to create a new issue/PR regarding the same.